File: examples/go/calculator_client.go
package main

import (
	"context"
	"fmt"

	"github.com/filecoin-project/go-jsonrpc"
)

func main() {
	var client struct {
		Add      func(ctx context.Context, params map[string]interface{}) (int, error)
		Subtract func(ctx context.Context, params map[string]interface{}) (int, error)
		Multiply func(ctx context.Context, params map[string]interface{}) (int, error)
		Divide   func(ctx context.Context, params map[string]interface{}) (float64, error)
	}

	closer, err := jsonrpc.NewClient(context.Background(), "http://localhost:8080", "Calculator", &client, nil)
	if err != nil {
		panic(err)
	}
	defer closer()

	addParams := map[string]interface{}{
		"a": 10,
		"b": 5,
	}
	addResult, err := client.Add(context.Background(), addParams)
	if err != nil {
		panic(err)
	}
	fmt.Printf("10 + 5 = %d\n", addResult)

	subtractParams := map[string]interface{}{
		"a": 10,
		"b": 5,
	}
	subtractResult, err := client.Subtract(context.Background(), subtractParams)
	if err != nil {
		panic(err)
	}
	fmt.Printf("10 - 5 = %d\n", subtractResult)

	multiplyParams := map[string]interface{}{
		"a": 10,
		"b": 5,
	}
	multiplyResult, err := client.Multiply(context.Background(), multiplyParams)
	if err != nil {
		panic(err)
	}
	fmt.Printf("10 * 5 = %d\n", multiplyResult)

	divideParams := map[string]interface{}{
		"a": 10,
		"b": 5,
	}
	divideResult, err := client.Divide(context.Background(), divideParams)
	if err != nil {
		panic(err)
	}
	fmt.Printf("10 / 5 = %.1f\n", divideResult)
}

File: examples/calculator_client.py
from easy_jsonrpc import EasyJSONRPCClient


def main():
    client = EasyJSONRPCClient("http://localhost:8080")

    try:
        add_result = client.call("Calculator.Add", {"a": 10, "b": 5})
        print(f"10 + 5 = {add_result}")
    except Exception as e:
        print(f"Add 호출 중 오류 발생: {e}")

    try:
        subtract_result = client.call("Calculator.Subtract", {"a": 10, "b": 5})
        print(f"10 - 5 = {subtract_result}")
    except Exception as e:
        print(f"Subtract 호출 중 오류 발생: {e}")

    try:
        multiply_result = client.call("Calculator.Multiply", {"a": 10, "b": 5})
        print(f"10 * 5 = {multiply_result}")
    except Exception as e:
        print(f"Multiply 호출 중 오류 발생: {e}")

    try:
        divide_result = client.call("Calculator.Divide", {"a": 10, "b": 5})
        print(f"10 / 5 = {divide_result}")
    except Exception as e:
        print(f"Divide 호출 중 오류 발생: {e}")

    try:
        client.notify("Calculator.Add", {"a": 10, "b": 5})
        print(f"알림보내기 완료")
    except Exception as e:
        print(f"Add 호출 중 오류 발생: {e}")


if __name__ == "__main__":
    main()
